### ISSUE TITLE
BO - Shipping preferences - ID of the carrier is displayed in the  Default carrier list after fresh install

### ISSUE THREAD

**Issue by khouloudbelguith** [PrestaShop/PrestaShop#28878]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

This issue id detected by nightly: https://nightly.prestashop.com/report/4307
ping @nesrineabdmouleh @boubkerbribri 
Related to this PR: https://github.com/PrestaShop/PrestaShop/pull/28718
In the BO > Shipping > Preferences page > Carrier options, the id of the carrier is always displayed.
![image](https://user-images.githubusercontent.com/16067358/175988543-5c5a2c37-2344-4205-a55e-78926448cfe7.png)

### Expected behavior

The id should be displayed only when we have two carriers with the same name
![image](https://user-images.githubusercontent.com/16067358/175989041-c14ea512-a940-4d6e-b382-d741f1a079e2.png)

### Steps to reproduce

1. Fresh install develop
2. Go to BO > Shipping > Preferences page > Carrier options
3. Check the list > the id of the carrier is displayed although we don't have two carriers with the same name

![image](https://user-images.githubusercontent.com/16067358/175988927-212addc6-d15b-42ba-8e50-38ae65dd8a8d.png)

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

8.1

**Comment by ghost** [comment]
Hello, @khouloudbelguith 

it is the expected behavior of the PR

**Comment by MatShir** [comment]
You are right, I thought It was a regression because usually in the back-office the ID is shown only when there are 2 same items in the list.

**Comment by MatShir** [comment]
So, we will adapt the auto-test to the scope of the initial PR but to keep consistency in the BO, I will keep this issue open so at some point we will adjust the behavior and only display the id when needed.
